core/tests/webdriverio_utils/BlogPages.js

```diff
@@ -0,0 +1,308 @@
+  var paginationNextButton = $('.e2e-test-pagination-next-button');
+  var paginationPrevButton = $('.e2e-test-pagination-prev-button');
+  var paginationContainer = $('.e2e-test-pagination');
+  var searchInput = $('.e2e-test-search-input');
+  var blogPostTilesSelector = function() {
+    return $$('.e2e-test-blog-post-tile-item');
+  };
+  var blogPostRecommendationTilesSelector = function() {
+    return $$('.e2e-test-blog-post-tile-item');
+  };
+
+  this.submitSearchQuery = async function(searchQuery) {
+    await waitFor.pageToFullyLoad();
+    await action.clear('Search input', searchInput);
+    await action.setValue('Search input', searchInput, searchQuery);
+  };
+
+  this.get = async function() {
+    await browser.url(general.BLOG_PAGE_URL_SUFFIX);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.getBlogPostSearchPage = async function(searchQuery) {
+    await browser.url(
+      general.BLOG_PAGE_SEARCH_URL_PREFIX + '?q=' + searchQuery);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.expectNoResultsFoundShown = async function() {
+    await this.waitForVisibilityOfBlogHomePageContainer();
+    await waitFor.visibilityOf(
+      noResultsFoundContainer,
+      'No results found container taking too long to display');
+  };
+
+  this.expectBlogHomePageWelcomeHeadingToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      blogHomepageWelcomeHeading,
+      'Blog Home Page Heading taking too long to display');
+  };
+
+  this.expectOppiaAvatarImageToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      oppiaAvatarImage,
+      'Oppia Image Avatar image taking too long to display');
+  };
+
+  this.expectTagFilterComponentToBeVisible = async function() {
+    await waitFor.visibilityOf(
+      blogPostTagFilter,
+      'Tag Filter component taking too long to display');
+  };
+
+  this.getBlogDashboardPage = async function() {
+    blogDashboardPage = new BlogDashboardPage.BlogDashboardPage();
+    await blogDashboardPage.get();
+    await blogDashboardPage.expectCurrUserToHaveProfilePhoto();
+  };
+
+  this.publishNewBlogPostFromBlogDashboard = async function(
+      blogPostTitle, richTextContent, tagsList
+  ) {
+    await blogDashboardPage.createNewBlogPost();
+    await blogDashboardPage.publishNewBlogPost(
+      blogPostTitle,
+      await forms.toRichText(richTextContent),
+      tagsList
+    );
+    await blogDashboardPage.navigateToBlogDashboardPageWithBackButton();
+  };
+
+  this.saveBlogPostAsDraftFromBlogDashboard = async function(
+      blogPostTitle, richTextContent
+  ) {
+    await blogDashboardPage.createNewBlogPost();
+    await blogDashboardPage.saveBlogPostAsDraft(
+      blogPostTitle,
+      await forms.toRichText(richTextContent),
+    );
+    await blogDashboardPage.navigateToBlogDashboardPageWithBackButton();
+  };
+
+  this.expectBlogPostSearchFieldToBeVisible = async function() {
+    expect(await blogPostSearchField.isDisplayed()).toBe(true);
```

Comment:
Can you use a waitFor here instead to handle cases where it takes some time for the field to load?

Reply:
Done